src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -34,8 +35,12 @@ public ActionResult execute() throws EntityDoesNotExistException {
         
         String confirmUrl = Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED 
                 + "?" + Const.ParamsNames.REGKEY + "=" + regkey 
-                + "&" + Const.ParamsNames.NEXT_URL + "=" + nextUrl;
-        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl, Logic.getLogoutUrl(confirmUrl));
+                + "&" + Const.ParamsNames.NEXT_URL + "=" + Sanitizer.sanitizeForNextUrl(nextUrl);
+        String studentNewCourse = getRequestParamValue(Const.ParamsNames.STUDENT_NEW_COURSE);
+        boolean isUserLoggedInAsAnotherStudent = studentNewCourse != null;
+        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl,
+                                                         Logic.getLogoutUrl(Sanitizer.sanitizeForNextUrl(confirmUrl)),
+                                                         isUserLoggedInAsAnotherStudent, studentNewCourse);
```

Comment:
two things here:
1. the name ```isUserLoggedInAsAnotherStudent``` is confusing
2. the course that we are sending in as a parameter is the same as ```student.course``` which is obtained from the regkey in the registration link.

I understand that you are trying to fix the problem by adding a new parameter that indicates whether a special action is needed or not. I believe the simplest solution is what you had done earlier, but instead of checking if the next_url is in the ```PAGES_ACCESSIBLE_WITHOUT_GOOGLE_LOGIN``` list, check if the next_url is NOT IN ```PAGES_ACCESSIBLE_WITHOUT_REGISTRATION```. This is future-proof and technically this is what we should be targeting. 

Just my thoughts...